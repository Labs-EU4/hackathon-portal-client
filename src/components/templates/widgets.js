import React, { useEffect, useRef } from "react";
import Button from "../atoms/Button";
import { RowBody } from "../atoms/RowBody";
import { useHistory } from "react-router-dom";
import {
  StyledContainer,
  Container,
  StyledWidget
} from "../styles/templates/AppParticipantTeams";

export const UserWidget = ({ user, select, ...otherProps }) => {
  return (
    <StyledWidget key={user.id} onClick={() => select(user)} {...otherProps}>
      {user.email}
    </StyledWidget>
  );
};

export const SearchWidget = props => {
  const history = useHistory();

  const redirect = (location = "/dashboard") => {
    history.push(location);
  };
  const searchString = props.searchString;
  const setSearchString = props.setSearchString;
  const inputRef = useRef(null);
  useEffect(() => {
    inputRef.current.focus();
  }, []);

  return (
    <Container display="wide">
      <input
        type="text"
        value={searchString}
        onChange={e => {
          setSearchString(e.target.value);
        }}
        placeholder="Search by email"
        ref={inputRef}
      />
      {props.matches.map(user => (
        <UserWidget key={user.id} user={user} select={props.setSelectedUser} />
      ))}
      {!!props.matches && props.validateEmail(searchString)
        ? props.setNoneUser(searchString)
        : props.setNoneUser(null)}
      <Button color="grey" onClick={() => redirect(history)}>
        Back to dashboard
      </Button>
    </Container>
  );
};
export const RoleWidget = props => {
  const selectedUser = props.selectedUser;
  const handleSubmit = props.handleSubmit;
  const history = useHistory();

  const redirect = (location = "/dashboard") => {
    history.push(location);
  };
  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          You are adding{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {selectedUser.email}
          </span>{" "}
          to your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="grey" onClick={() => redirect()}>
          Back to dashboard
        </Button>
        <Button
          color="green"
          onClick={selectedUser => handleSubmit(selectedUser)}
        >
          Add teammate
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

export const InviteWidget = props => {
  const noneUser = props.noneUser;
  const sendInvite = props.sendInvite;

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          This user is not on this platform. click send to invite{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {noneUser}
          </span>{" "}
          to join your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="green" onClick={noneUser => sendInvite(noneUser)}>
          Send Invite
        </Button>
      </RowBody>
    </StyledContainer>
  );
};
