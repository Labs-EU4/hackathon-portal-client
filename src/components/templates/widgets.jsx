import React, { useEffect, useRef } from "react";
import { useHistory } from "react-router-dom";
import styled from 'styled-components';

import Button from "../atoms/Button";
import { RowBody } from "../../assets/styles/atoms/RowBodyStyling";
import { useSearchUserByEmail } from "../../hooks";
import {
  StyledContainer,
  AddTeamParticipantContainer,
  StyledWidget
} from "../../assets/styles/templates/AddParticipantTeamsStyling";
import { 
  Container,
  ContainerRadio,
  UsersList,
  ChosenJudgesContainer,
  ChosenJudgeImg,
  StyledSearchIcon,
  StyledJudgeWidget,
  UserAvatar,
  UserInfo
} from "../../assets/styles/templates/AddTeammatesStyling";
import isEmail from "validator/lib/isEmail";


const UserWidget = ({ user, select, ...otherProps }) => {
  const StyledWidget = styled.div`
    margin-bottom: 10px;
    cursor: pointer;
  `;
  return (
    <StyledWidget key={user.id} onClick={() => select(user)} {...otherProps}>
      {user.email}
    </StyledWidget>
  );
};

export const SearchWidget = props => {
  const { setSelectedUser, setNoneUser } = props;
  const history = useHistory();
  const [matches, searchString, setSearchString] = useSearchUserByEmail();
  const validateEmail = email => {
    return isEmail(email);
  };
  const redirect = (location = "/dashboard") => {
    history.push(location);
  };
  const inputRef = useRef(null);
  useEffect(() => {
    inputRef.current.focus();
  }, []);

  return (
    <AddTeamParticipantContainer display="wide">
      <input
        type="text"
        value={searchString}
        onChange={e => {
          setSearchString(e.target.value);
        }}
        placeholder="Search by email"
        ref={inputRef}
      />
      {matches.map(user => (
        <UserWidget key={user.id} user={user} select={setSelectedUser} />
      ))}
      {!matches && validateEmail(searchString)
        ? setNoneUser(searchString)
        : setNoneUser(null)}
      <Button 
        color="grey" 
        onClick={() => redirect(history)}>
        Back to dashboard
      </Button>
    </AddTeamParticipantContainer>
  );
};

export const ParticipantRoleWidget = props => {
  const { selectedUser, handleSubmit} = props;
  const history = useHistory();
  const redirect = (location = "/dashboard") => {
    history.push(location);
  };

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          You are adding{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {selectedUser.email}
          </span>{" "}
          to your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="grey" onClick={() => redirect()}>
          Back to dashboard
        </Button>
        <Button
          color="green"
          onClick={selectedUser => handleSubmit(selectedUser)}
        >
          Add teammate
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

export const ParticipantInviteWidget = props => {
  const noneUser = props.noneUser;
  const sendInvite = props.sendInvite;

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          This user is not on this platform. click send to invite{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {noneUser}
          </span>{" "}
          to join your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="green" onClick={noneUser => sendInvite(noneUser)}>
          Send Invite
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

const Radio = ({ label, value, type = "radio", ...radioProps }) => {
  return (
    <ContainerRadio>
      {label || value}
      <input type={type} {...radioProps} />
      <span></span>
    </ContainerRadio>
  );
};

export const TeamRoleWidget = (props) => {
  const setRole = props.setRole;
  const role = props.role
  const handleSubmit = props.handleSubmit;
  const history = useHistory();
  const redirect = (location = "/dashboard") => {
    history.push(location);
  };

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <Radio
          label="organizer"
          name="role"
          onChange={() => setRole("organizer")}
          checked={role === "organizer"}
        />
        <Radio
          name="role"
          label="judge"
          onChange={() => setRole("judge")}
          checked={role === "judge"}
        />
      </RowBody>
      <RowBody>
        <Button color="grey" onClick={() => redirect()}>
          Back to dashboard
        </Button>
        <Button color="green" onClick={handleSubmit}>
          Add teammate
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

export const TeamInviteWidget = props => {
  const noneUser = props.noneUser;
  const sendInvite = props.sendInvite;
  const setRole = props.setRole;
  const role = props.role

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          This user is not on this platform. Please select a role for
        click send to invite {" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {noneUser}
          </span>{" "}
          to join your team
      </h6>
      </RowBody>
      <RowBody direction="column-reverse">
        <Radio
          label="organizer"
          name="role"
          onChange={() => setRole("organizer")}
          checked={role === "organizer"}
        />
        <Radio
          name="role"
          label="judge"
          onChange={() => setRole("judge")}
          checked={role === "judge"}
        />
      </RowBody>
      <RowBody>
        <Button color="green" onClick={sendInvite}>
          Send Invite
      </Button>
      </RowBody>
    </StyledContainer>
  );
};

export const JudgesSearchWidget = props => {
  const { 
    selectedUsersHandler, 
    selectedUserArr,
    setNoneUser,
    handleExit,
    handleSubmit,
  } = props;
  const [matches, searchString, setSearchString] = useSearchUserByEmail();
  const validateEmail = email => {
    return isEmail(email);
  };
  const inputRef = useRef(null);

  useEffect(() => {
    inputRef.current.focus();
  }, []);

  return (
    <Container display="wide">
      <input
        type="text"
        value={searchString}
        onChange={e => {
          setSearchString(e.target.value);
        }}
        placeholder="Search by email"
        ref={inputRef}
      />
      {
        searchString.length > 0 
          ? <StyledSearchIcon icon="times" onClick={() => setSearchString('')}/> 
          : <StyledSearchIcon icon="search" />
      }
      <div style={{ display: "flex", flexDirection: "column"}}>
        <UsersList>
          {matches.map(user => (
            // <UserWidget key={user.id} user={user} select={setSelectedUser} />
            <JudgeWidget key={user.id} user={user} selected={selectedUsersHandler} />
          ))}
          {
            !!matches && validateEmail(searchString) 
              ? setNoneUser(searchString) 
              : setNoneUser(null)
          }
        </UsersList>
        <ChosenJudgesContainer>
          {
            selectedUserArr.current.length > 0 && (
              selectedUserArr.current.map(user => {
                let memberProfile = JSON.parse(user.image_url);
                return user.image_url !== null ? (
                  <ChosenJudgeImg src={memberProfile.avatar} alt={user.username}/>
                ) : (
                  <ChosenJudgeImg 
                    src="https://media.giphy.com/media/g0QET2Iejaa4EQ0eBV/giphy.gif" alt="default-img" 
                  />
                )
              }) 
            )
          }
        </ChosenJudgesContainer>
      </div>
      <RowBody>
        <Button 
          color="grey" 
          size="half" 
          onClick={handleExit}
        >Back</Button>
        <Button 
          color="green" 
          size="half" 
          onClick={handleSubmit}
        >Add Judge</Button>
      </RowBody>
    </Container>
  );
};

const JudgeWidget = ({ user, selected, ...otherProps }) => {
  let memberProfile = JSON.parse(user.image_url);

  const selectedJudgeHandler = (e) => {
    e.currentTarget.classList.add('selected');
    // Add option to deselect user with conditional to check for that
    selected(user);
  }

  return (
    <StyledJudgeWidget 
      key={user.id} 
      // onClick={() => selected(user)} 
      onClick={selectedJudgeHandler} 
      {...otherProps}
    >
      <UserAvatar>
        {
          user.image_url !== null ? (
            <img src={memberProfile.avatar} alt={user.username}/>
          ) : (
            <img 
              src="https://media.giphy.com/media/g0QET2Iejaa4EQ0eBV/giphy.gif" alt="default-img" 
            />
          )
        }
      </UserAvatar>
      <UserInfo>
        <p>{user.username}</p>
        <p>{user.email}</p>
      </UserInfo>
    </StyledJudgeWidget>
  );
};