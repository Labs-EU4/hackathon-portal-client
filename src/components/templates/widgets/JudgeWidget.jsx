import React from 'react';

import {
    StyledJudgeWidget,
    UserAvatar,
    UserInfo
} from "../../assets/styles/templates/AddTeammatesStyling";

export const JudgeWidget = ({ user, selected, ...otherProps }) => {
    let memberProfile = JSON.parse(user.image_url);
  
    const selectedJudgeHandler = (e) => {
      e.currentTarget.classList.add('selected');
      // Add option to deselect user with conditional to check for that
      selected(user);
    }
  
    return (
      <StyledJudgeWidget 
        key={user.id} 
        // onClick={() => selected(user)} 
        onClick={selectedJudgeHandler} 
        {...otherProps}
      >
        <UserAvatar>
          {
            user.image_url !== null ? (
              <img src={memberProfile.avatar} alt={user.username}/>
            ) : (
              <img 
                src="https://media.giphy.com/media/g0QET2Iejaa4EQ0eBV/giphy.gif" alt="default-img" 
              />
            )
          }
        </UserAvatar>
        <UserInfo>
          <p>{user.username}</p>
          <p>{user.email}</p>
        </UserInfo>
      </StyledJudgeWidget>
    );
};