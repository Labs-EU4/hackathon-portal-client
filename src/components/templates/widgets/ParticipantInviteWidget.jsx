import React from 'react';

export const ParticipantInviteWidget = props => {
    const noneUser = props.noneUser;
    const sendInvite = props.sendInvite;
  
    return (
      <StyledContainer>
        <RowBody direction="column-reverse">
          <h6>
            This user is not on this platform. click send to invite{" "}
            <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
              {noneUser}
            </span>{" "}
            to join your team
          </h6>
        </RowBody>
        <RowBody>
          <Button color="green" onClick={noneUser => sendInvite(noneUser)}>
            Send Invite
          </Button>
        </RowBody>
      </StyledContainer>
    );
};