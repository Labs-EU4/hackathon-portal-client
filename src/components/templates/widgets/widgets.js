import React, { useEffect, useRef } from "react";
import { useHistory } from "react-router-dom";
import styled from 'styled-components';

import Button from "../atoms/Button";
import { RowBody } from "../../assets/styles/atoms/RowBodyStyling";
import { useSearchUserByEmail } from "../../hooks";
import {
  StyledContainer,
  AddTeamParticipantContainer,
  StyledWidget
} from "../../assets/styles/templates/AddParticipantTeamsStyling";
import { 
  Container,
  ContainerRadio
} from "../../assets/styles/templates/AddTeammatesStyling";
import isEmail from "validator/lib/isEmail";


const UserWidget = ({ user, select, ...otherProps }) => {
  const StyledWidget = styled.div`
    margin-bottom: 10px;
    cursor: pointer;
  `;
  return (
    <StyledWidget key={user.id} onClick={() => select(user)} {...otherProps}>
      {user.email}
    </StyledWidget>
  );
};

export const SearchWidget = props => {
  const { setSelectedUser, setNoneUser } = props;
  const history = useHistory();
  const [matches, searchString, setSearchString] = useSearchUserByEmail();
  const validateEmail = email => {
    return isEmail(email);
  };
  const redirect = (location = "/dashboard") => {
    history.push(location);
  };
  const inputRef = useRef(null);
  useEffect(() => {
    inputRef.current.focus();
  }, []);

  return (
    <AddTeamParticipantContainer display="wide">
      <input
        type="text"
        value={searchString}
        onChange={e => {
          setSearchString(e.target.value);
        }}
        placeholder="Search by email"
        ref={inputRef}
      />
      {matches.map(user => (
        <UserWidget key={user.id} user={user} select={setSelectedUser} />
      ))}
      {!matches && validateEmail(searchString)
        ? setNoneUser(searchString)
        : setNoneUser(null)}
      <Button 
        color="grey" 
        onClick={() => redirect(history)}>
        Back to dashboard
      </Button>
    </AddTeamParticipantContainer>
  );
};

export const ParticipantRoleWidget = props => {
  const { selectedUser, handleSubmit} = props;
  const history = useHistory();
  const redirect = (location = "/dashboard") => {
    history.push(location);
  };

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          You are adding{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {selectedUser.email}
          </span>{" "}
          to your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="grey" onClick={() => redirect()}>
          Back to dashboard
        </Button>
        <Button
          color="green"
          onClick={selectedUser => handleSubmit(selectedUser)}
        >
          Add teammate
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

export const ParticipantInviteWidget = props => {
  const noneUser = props.noneUser;
  const sendInvite = props.sendInvite;

  return (
    <StyledContainer>
      <RowBody direction="column-reverse">
        <h6>
          This user is not on this platform. click send to invite{" "}
          <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
            {noneUser}
          </span>{" "}
          to join your team
        </h6>
      </RowBody>
      <RowBody>
        <Button color="green" onClick={noneUser => sendInvite(noneUser)}>
          Send Invite
        </Button>
      </RowBody>
    </StyledContainer>
  );
};

const Radio = ({ label, value, type = "radio", ...radioProps }) => {
  return (
    <ContainerRadio>
      {label || value}
      <input type={type} {...radioProps} />
      <span></span>
    </ContainerRadio>
  );
};