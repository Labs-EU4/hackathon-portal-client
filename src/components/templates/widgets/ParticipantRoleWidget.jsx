import React from 'react';

import Button from "../atoms/Button";
import { RowBody } from "../../assets/styles/atoms/RowBodyStyling";
import { StyledContainer } from "../../assets/styles/templates/AddParticipantTeamsStyling";

export const ParticipantRoleWidget = props => {
    const { selectedUser, handleSubmit} = props;
    const history = useHistory();
    const redirect = (location = "/dashboard") => {
      history.push(location);
    };
  
    return (
      <StyledContainer>
        <RowBody direction="column-reverse">
          <h6>
            You are adding{" "}
            <span style={{ color: "#273F92", backgroundColor: "aliceblue" }}>
              {selectedUser.email}
            </span>{" "}
            to your team
          </h6>
        </RowBody>
        <RowBody>
          <Button color="grey" onClick={() => redirect()}>
            Back to dashboard
          </Button>
          <Button
            color="green"
            onClick={selectedUser => handleSubmit(selectedUser)}
          >
            Add teammate
          </Button>
        </RowBody>
      </StyledContainer>
    );
};