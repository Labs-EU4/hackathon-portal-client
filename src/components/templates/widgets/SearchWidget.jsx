import React from 'react';

import Button from "../../atoms/Button";
import { AddTeamParticipantContainer } from "../../../assets/styles/templates/AddParticipantTeamsStyling";
import isEmail from "validator/lib/isEmail";
import { useSearchUserByEmail } from "../../../hooks";

export const SearchWidget = props => {
    const { setSelectedUser, setNoneUser } = props;
    const history = useHistory();
    const [matches, searchString, setSearchString] = useSearchUserByEmail();
    const validateEmail = email => {
      return isEmail(email);
    };
    const redirect = (location = "/dashboard") => {
      history.push(location);
    };
    const inputRef = useRef(null);
    useEffect(() => {
      inputRef.current.focus();
    }, []);
  
    return (
      <AddTeamParticipantContainer display="wide">
        <input
          type="text"
          value={searchString}
          onChange={e => {
            setSearchString(e.target.value);
          }}
          placeholder="Search by email"
          ref={inputRef}
        />
        {matches.map(user => (
          <UserWidget key={user.id} user={user} select={setSelectedUser} />
        ))}
        {!matches && validateEmail(searchString)
          ? setNoneUser(searchString)
          : setNoneUser(null)}
        <Button 
          color="grey" 
          onClick={() => redirect(history)}>
          Back to dashboard
        </Button>
      </AddTeamParticipantContainer>
    );
};