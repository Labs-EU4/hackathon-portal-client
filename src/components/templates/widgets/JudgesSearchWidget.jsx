import React, { useEffect, useRef } from 'react';

import Button from "../../atoms/Button";
import { RowBody } from "../../../assets/styles/atoms/RowBodyStyling";
import { 
    Container,
    UsersList,
    ChosenJudgesContainer,
    ChosenJudgeImg,
    StyledSearchIcon
} from "../../../assets/styles/templates/AddTeammatesStyling";
import { JudgeWidget } from './JudgeWidget';
import isEmail from "validator/lib/isEmail";
import { useSearchUserByEmail } from "../../../hooks";

export const JudgesSearchWidget = props => {
    const { 
      selectedUsersHandler, 
      selectedUserArr,
      setNoneUser,
      handleExit,
      handleSubmit,
    } = props;
    const [matches, searchString, setSearchString] = useSearchUserByEmail();
    const validateEmail = email => {
      return isEmail(email);
    };
    const inputRef = useRef(null);
  
    useEffect(() => {
      inputRef.current.focus();
    }, []);
  
    return (
      <Container display="wide">
        <input
          type="text"
          value={searchString}
          onChange={e => {
            setSearchString(e.target.value);
          }}
          placeholder="Search by email"
          ref={inputRef}
        />
        {
          searchString.length > 0 
            ? <StyledSearchIcon icon="times" onClick={() => setSearchString('')}/> 
            : <StyledSearchIcon icon="search" />
        }
        <div style={{ display: "flex", flexDirection: "column"}}>
          <UsersList>
            {matches.map(user => (
              // <UserWidget key={user.id} user={user} select={setSelectedUser} />
              <JudgeWidget key={user.id} user={user} selected={selectedUsersHandler} />
            ))}
            {
              !!matches && validateEmail(searchString) 
                ? setNoneUser(searchString) 
                : setNoneUser(null)
            }
          </UsersList>
          <ChosenJudgesContainer>
            {
              selectedUserArr.current.length > 0 && (
                selectedUserArr.current.map(user => {
                  let memberProfile = JSON.parse(user.image_url);
                  return user.image_url !== null ? (
                    <ChosenJudgeImg src={memberProfile.avatar} alt={user.username}/>
                  ) : (
                    <ChosenJudgeImg 
                      src="https://media.giphy.com/media/g0QET2Iejaa4EQ0eBV/giphy.gif" alt="default-img" 
                    />
                  )
                }) 
              )
            }
          </ChosenJudgesContainer>
        </div>
        <RowBody>
          <Button 
            color="grey" 
            size="half" 
            onClick={handleExit}
          >Back</Button>
          <Button 
            color="green" 
            size="half" 
            onClick={handleSubmit}
          >Add Judge</Button>
        </RowBody>
      </Container>
    );
};