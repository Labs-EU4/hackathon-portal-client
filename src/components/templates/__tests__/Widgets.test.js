import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import { initialState } from "../../../utils/mockData";
import { ParticipantInviteWidget, TeamRoleWidget } from "../widgets";
import { theme } from "../../../assets/styles/ThemeStyling";
import { ThemeProvider } from "styled-components";
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
import { far } from '@fortawesome/free-regular-svg-icons';
import { fab } from '@fortawesome/free-brands-svg-icons';

library.add(fas, far, fab);

const history = createMemoryHistory();

afterEach(cleanup);

let jestFeatures;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  jestFeatures = render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <ParticipantInviteWidget />
          <TeamRoleWidget />
        </ThemeProvider>
      </Provider>
    </Router>
  );
});

describe("Component participantinvitewidget renders the text correctly", () => {
  it("the h6 tag within the component is rendering properly", () => {
    expect(jestFeatures.queryByText(/This user/i)).toBeInTheDocument();
  });
  it("renders without crashing ", () => {
    expect(jestFeatures).toMatchSnapshot();
  });
});

describe("Component TeamRoleWidget renders the text correctly", () => {
  it("the button within the component is rendering properly", () => {
    expect(jestFeatures.queryByText(/Back/i)).toBeInTheDocument();
  });
});
