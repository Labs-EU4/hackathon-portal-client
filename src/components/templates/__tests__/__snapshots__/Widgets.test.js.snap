// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component participantinvitewidget renders the text correctly renders without crashing  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gzVnrw jJOFkB"
      >
        <div
          class="sc-bxivhb GFwqS"
          direction="column-reverse"
        >
          <h6>
            This user is not on this platform. click send to invite
             
            <span
              style="color: rgb(39, 63, 146); background-color: aliceblue;"
            />
             
            to join your team
          </h6>
        </div>
        <div
          class="sc-bxivhb gBMiwD"
        >
          <button
            class="sc-bdVaJa gFEKGJ"
            color="green"
          >
            Send Invite
          </button>
        </div>
      </div>
      <div
        class="sc-gzVnrw jJOFkB"
      >
        <div
          class="sc-bxivhb GFwqS"
          direction="column-reverse"
        >
          <label
            class="sc-kGXeez eriyDj"
          >
            organizer
            <input
              name="role"
              type="radio"
            />
            <span />
          </label>
          <label
            class="sc-kGXeez eriyDj"
          >
            judge
            <input
              name="role"
              type="radio"
            />
            <span />
          </label>
        </div>
        <div
          class="sc-bxivhb gBMiwD"
        >
          <button
            class="sc-bdVaJa fcBjIo"
            color="grey"
          >
            Back to dashboard
          </button>
          <button
            class="sc-bdVaJa gFEKGJ"
            color="green"
          >
            Add teammate
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gzVnrw jJOFkB"
    >
      <div
        class="sc-bxivhb GFwqS"
        direction="column-reverse"
      >
        <h6>
          This user is not on this platform. click send to invite
           
          <span
            style="color: rgb(39, 63, 146); background-color: aliceblue;"
          />
           
          to join your team
        </h6>
      </div>
      <div
        class="sc-bxivhb gBMiwD"
      >
        <button
          class="sc-bdVaJa gFEKGJ"
          color="green"
        >
          Send Invite
        </button>
      </div>
    </div>
    <div
      class="sc-gzVnrw jJOFkB"
    >
      <div
        class="sc-bxivhb GFwqS"
        direction="column-reverse"
      >
        <label
          class="sc-kGXeez eriyDj"
        >
          organizer
          <input
            name="role"
            type="radio"
          />
          <span />
        </label>
        <label
          class="sc-kGXeez eriyDj"
        >
          judge
          <input
            name="role"
            type="radio"
          />
          <span />
        </label>
      </div>
      <div
        class="sc-bxivhb gBMiwD"
      >
        <button
          class="sc-bdVaJa fcBjIo"
          color="grey"
        >
          Back to dashboard
        </button>
        <button
          class="sc-bdVaJa gFEKGJ"
          color="green"
        >
          Add teammate
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
